diff -Naur exempi-2.2.0/exempi/Makefile.am exempi-2.2.0-nemo-testing/exempi/Makefile.am
--- exempi-2.2.0/exempi/Makefile.am	2011-11-13 06:57:42.000000000 +0000
+++ exempi-2.2.0-nemo-testing/exempi/Makefile.am	2013-04-29 13:57:00.690877766 +0000
@@ -34,7 +34,7 @@
 # OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 
-SUBDIRS = tests
+SUBDIRS = . tests
 DIST_SUBDIRS = tests doc
 
 exempidir = $(includedir)/@EXEMPI_INCLUDE_BASE@/exempi
diff -Naur exempi-2.2.0/exempi/tests/Makefile.am exempi-2.2.0-nemo-testing/exempi/tests/Makefile.am
--- exempi-2.2.0/exempi/tests/Makefile.am	2011-11-13 06:57:42.000000000 +0000
+++ exempi-2.2.0-nemo-testing/exempi/tests/Makefile.am	2013-04-29 14:04:50.880273549 +0000
@@ -54,7 +54,6 @@
 EXTRA_DIST = $(check_DATA) $(check_SCRIPTS)
 CLEANFILES = test.jpg
 
-
 testinit_SOURCES = testinit.cpp utils.cpp
 testinit_LDADD = ../libexempi.la @BOOST_UNIT_TEST_FRAMEWORK_LIBS@
 testinit_LDFLAGS = -static @BOOST_UNIT_TEST_FRAMEWORK_LDFLAGS@
@@ -93,4 +92,11 @@
 
 testcpp_SOURCES = testcpp.cpp utils.cpp
 testcpp_LDADD = ../libexempi.la @BOOST_UNIT_TEST_FRAMEWORK_LIBS@
-testcpp_LDFLAGS = -static @BOOST_UNIT_TEST_FRAMEWORK_LDFLAGS@
\ No newline at end of file
+testcpp_LDFLAGS = -static @BOOST_UNIT_TEST_FRAMEWORK_LDFLAGS@
+
+nemotestsbindir = /opt/tests/exempi/bin
+nemotestsdatadir = /opt/tests/exempi/data
+
+nemotestsbin_PROGRAMS = $(check_PROGRAMS) 
+nemotestsdata_DATA = test1.xmp fdo18635.jpg
+
diff -Naur exempi-2.2.0/exempi/tests/test-tiff-leak.cpp exempi-2.2.0-nemo-testing/exempi/tests/test-tiff-leak.cpp
--- exempi-2.2.0/exempi/tests/test-tiff-leak.cpp	2011-11-13 06:57:42.000000000 +0000
+++ exempi-2.2.0-nemo-testing/exempi/tests/test-tiff-leak.cpp	2013-04-30 06:20:45.601155973 +0000
@@ -55,8 +55,9 @@
 {
 	prepare_test(argc, argv, "../../samples/testfiles/BlueSquare.jpg");
 
-	std::string orig_tiff_file = g_src_testdir 
-		+ "../../samples/testfiles/BlueSquare.tif";
+	//std::string orig_tiff_file = g_src_testdir 
+	//	+ "../../samples/testfiles/BlueSquare.tif";
+	std::string orig_tiff_file = "/opt/tests/exempi/data/BlueSquare.tif";
 	BOOST_CHECK(copy_file(orig_tiff_file, "test.tif"));
 	BOOST_CHECK(chmod("test.tif", S_IRUSR|S_IWUSR) == 0);
 	BOOST_CHECK(xmp_init());
diff -Naur exempi-2.2.0/samples/Makefile.am exempi-2.2.0-nemo-testing/samples/Makefile.am
--- exempi-2.2.0/samples/Makefile.am	2011-11-13 06:57:42.000000000 +0000
+++ exempi-2.2.0-nemo-testing/samples/Makefile.am	2013-04-29 13:33:24.075603233 +0000
@@ -35,5 +35,5 @@
 #
 
 
-SUBDIRS = source
+SUBDIRS = source testfiles
 DIST_SUBDIRS = source testfiles
diff -Naur exempi-2.2.0/samples/source/Makefile.am exempi-2.2.0-nemo-testing/samples/source/Makefile.am
--- exempi-2.2.0/samples/source/Makefile.am	2012-01-22 07:49:31.000000000 +0000
+++ exempi-2.2.0-nemo-testing/samples/source/Makefile.am	2013-04-29 13:38:43.737565580 +0000
@@ -35,13 +35,16 @@
 #
 
 
-noinst_PROGRAMS = xmpcoverage xmpfilescoverage dumpxmp dumpmainxmp\
+noinst_PROGRAMS = xmpcoverage xmpfilescoverage dumpxmp\
 	customschema \
 	modifyingxmp \
 	readingxmp \
 	xmpcommandtool \
 	$(NULL)
 
+nemotestsbindir = /opt/tests/exempi/bin
+nemotestsbin_PROGRAMS = dumpmainxmp
+
 AM_CXXFLAGS = -fexceptions -funsigned-char -fPIC \
 	-Wno-multichar -Wno-ctor-dtor-privacy
 
@@ -85,4 +88,4 @@
 	xmpcommand/PrintUsage.cpp xmpcommand/PrintUsage.h \
 	xmpcommand/XMPCommand.cpp \
 	$(NULL)
-xmpcommandtool_LDADD = $(XMPLIBS) libsamplescommon.la
\ No newline at end of file
+xmpcommandtool_LDADD = $(XMPLIBS) libsamplescommon.la
diff -Naur exempi-2.2.0/samples/testfiles/Makefile.am exempi-2.2.0-nemo-testing/samples/testfiles/Makefile.am
--- exempi-2.2.0/samples/testfiles/Makefile.am	2011-11-13 06:57:42.000000000 +0000
+++ exempi-2.2.0-nemo-testing/samples/testfiles/Makefile.am	2013-04-29 13:42:30.208257700 +0000
@@ -37,3 +37,7 @@
 
 EXTRA_DIST=BlueSquare.ai   BlueSquare.eps   BlueSquare.gif BlueSquare.jpg  BlueSquare.mp3  BlueSquare.png  BlueSquare.tif\
 BlueSquare.avi  BlueSquare.indd  BlueSquare.mov  BlueSquare.pdf  BlueSquare.psd  BlueSquare.wav
+
+nemotestsdatadir = /opt/tests/exempi/data
+nemotestsdata_DATA = $(EXTRA_DIST)
+
