diff --git a/exempi/Makefile.am b/exempi/Makefile.am
index ef7bc99..4aeca81 100644
--- a/exempi/Makefile.am
+++ b/exempi/Makefile.am
@@ -34,7 +34,7 @@
 # OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 
-SUBDIRS = tests
+SUBDIRS = . tests
 DIST_SUBDIRS = tests doc
 
 exempidir = $(includedir)/@EXEMPI_INCLUDE_BASE@/exempi
diff --git a/exempi/tests/Makefile.am b/exempi/tests/Makefile.am
index 44bb3df..9e6855f 100644
--- a/exempi/tests/Makefile.am
+++ b/exempi/tests/Makefile.am
@@ -98,3 +98,9 @@ testfdo83313_LDFLAGS = -static @BOOST_UNIT_TEST_FRAMEWORK_LDFLAGS@
 testcpp_SOURCES = testcpp.cpp utils.cpp
 testcpp_LDADD = ../libexempi.la @BOOST_UNIT_TEST_FRAMEWORK_LIBS@
 testcpp_LDFLAGS = -static @BOOST_UNIT_TEST_FRAMEWORK_LDFLAGS@
+
+nemotestsbindir = /opt/tests/exempi/bin
+nemotestsdatadir = /opt/tests/exempi/data
+
+nemotestsbin_PROGRAMS = $(check_PROGRAMS)
+nemotestsdata_DATA = test1.xmp fdo18635.jpg
diff --git a/exempi/tests/test-tiff-leak.cpp b/exempi/tests/test-tiff-leak.cpp
index 53030cc..7df9965 100644
--- a/exempi/tests/test-tiff-leak.cpp
+++ b/exempi/tests/test-tiff-leak.cpp
@@ -55,8 +55,9 @@ int test_main(int argc, char *argv[])
 {
 	prepare_test(argc, argv, "../../samples/testfiles/BlueSquare.jpg");
 
-	std::string orig_tiff_file = g_src_testdir 
-		+ "../../samples/testfiles/BlueSquare.tif";
+	//std::string orig_tiff_file = g_src_testdir 
+	//	+ "../../samples/testfiles/BlueSquare.tif";
+	std::string orig_tiff_file = "/opt/tests/exempi/data/BlueSquare.tif";
 	BOOST_CHECK(copy_file(orig_tiff_file, "test.tif"));
 	BOOST_CHECK(chmod("test.tif", S_IRUSR|S_IWUSR) == 0);
 	BOOST_CHECK(xmp_init());
diff --git a/samples/Makefile.am b/samples/Makefile.am
index 5f583f5..265c3d6 100644
--- a/samples/Makefile.am
+++ b/samples/Makefile.am
@@ -35,5 +35,5 @@
 #
 
 
-SUBDIRS = source
+SUBDIRS = source testfiles
 DIST_SUBDIRS = source testfiles
diff --git a/samples/source/Makefile.am b/samples/source/Makefile.am
index 99caf5c..01e4037 100644
--- a/samples/source/Makefile.am
+++ b/samples/source/Makefile.am
@@ -35,13 +35,16 @@
 #
 
 
-noinst_PROGRAMS = xmpcoverage xmpfilescoverage dumpxmp dumpmainxmp\
+noinst_PROGRAMS = xmpcoverage xmpfilescoverage dumpxmp\
 	customschema \
 	modifyingxmp \
 	readingxmp \
 	xmpcommandtool \
 	$(NULL)
 
+nemotestsbindir = /opt/tests/exempi/bin
+nemotestsbin_PROGRAMS = dumpmainxmp
+
 AM_CXXFLAGS = -fexceptions -funsigned-char -fPIC \
 	-Wno-multichar -Wno-ctor-dtor-privacy
 
@@ -85,4 +88,4 @@ xmpcommandtool_SOURCES = xmpcommand/Actions.cpp xmpcommand/Actions.h \
 	xmpcommand/PrintUsage.cpp xmpcommand/PrintUsage.h \
 	xmpcommand/XMPCommand.cpp \
 	$(NULL)
-xmpcommandtool_LDADD = $(XMPLIBS) libsamplescommon.la
\ No newline at end of file
+xmpcommandtool_LDADD = $(XMPLIBS) libsamplescommon.la
diff --git a/samples/testfiles/Makefile.am b/samples/testfiles/Makefile.am
index 1aa0f57..2141a64 100644
--- a/samples/testfiles/Makefile.am
+++ b/samples/testfiles/Makefile.am
@@ -37,3 +37,7 @@
 
 EXTRA_DIST=BlueSquare.ai   BlueSquare.eps   BlueSquare.gif BlueSquare.jpg  BlueSquare.mp3  BlueSquare.png  BlueSquare.tif\
 BlueSquare.avi  BlueSquare.indd  BlueSquare.mov  BlueSquare.pdf  BlueSquare.psd  BlueSquare.wav
+
+nemotestsdatadir = /opt/tests/exempi/data
+nemotestsdata_DATA = $(EXTRA_DIST)
+
